### ISSUE TITLE
Can't set empty group

### ISSUE THREAD

**Issue by jtgasper3** [cannatag/ldap3#107]
I think I found a bug, but maybe it is user error.

I'm trying to create an add that replicates:

```
dn: cn=test2,ou=Groups,dc=example,dc=edu
changetype: add
objectClass: groupOfNames
objectClass: top
cn: test2
member: 
```

Having `member:` empty is valid for 389-ds which is used to create an empty `groupOfNames` object. I have not been able to replicate this using ldap3.

Using variations of:

```
.add(
            groupDn,
            object_class=['groupOfNames', 'top'],
            attributes={'cn': "test2", 'member': None},
            controls=None
        )
```
I've tried setting `member` to `None`, `''`, and `[]`.  They either cause TypeErrors or LDAP errors. Leaving the `member` attribute out all together causes an error because `member` is a required attribute.

Thoughts?

**Comment by cannatag** [comment]
hello, I can't test it now. In LDAP there is no idea of "empty" attributes. An empty attribute should not be shown at all. Probably the member attribute as a NULL value. Can you try with 'member':: [chr(0)] in the add operation?

Anyway are you sure that member can be empty? It it is defined as MUST in the schema it should have a value.

ANyway tomorrow I will test it in my lab.

Bye,
Giovanni

**Comment by jtgasper3** [comment]
Hi, 

I tested the `[chr(0)]` and no luck:

```
{'message': 'member: value #0 invalid per syntax', 'description': 'invalidAttributeSyntax', 'referrals': None, 'result': 21, 'type': 'addResponse', 'dn': ''}
```

I quickly spun up a second ldap server (I have Docker images for openLdap and 389-ds) and tried the same ldif and it also accepted it.

Apache Directory Studio exports the entry as well.

**Comment by jtgasper3** [comment]
I found a reference in http://www.ldapexplorer.com/en/manual/103021500-edit-set-empty-value.htm to setting an Empty value for member in a group. This looks to be user-agent specific, but the principal is the same.

Thanks for looking into this.

**Comment by cannatag** [comment]
Hi, please check code in dev. I've releaxed the attribute checking. Now you can send an empty attribute value (with '') even if this is forbidden by the rfc.

attributes={'cn': "test2", 'member': ''},


Let me know if it works for you.

Bye,
Giovanni

**Comment by jtgasper3** [comment]
Thanks for working on this. I'll probably not be able to test until Monday, but wanted to let you know that I am not ignoring you.

**Comment by jtgasper3** [comment]
Hello,

Using `attributes={'cn': "test2", 'member': ''},` worked perfectly for my case. Thanks!

Thanks for the attribution in the release notes... One side note... 389-ds also supports this empty  case.

**Comment by cannatag** [comment]
thanks. released in 0.9.9.1
